chef/cookbooks/network/recipes/default.rb

```diff
@@ -314,9 +334,76 @@ def kill_nic(nic)
+    if our_iface.ovs_master && our_iface.ovs_master.name != bridge
+      old_ovs = our_iface.ovs_master
+
+      # We are already attached to an ovs bridge, check whether that attachment
+      # was already verified (or created) during this chef-client run, by looking
+      # through the list of interfaces that we already touched ('ifs'). If it is,
+      # find out which crowbar network requires that attachement and error out
+      # with a meaningful error message.
+      if ifs.key? old_ovs.name
+        other_net = sorted_networks.find do |n|
+          n.bridge_name == old_ovs.name
+        end
+        other_net ||= sorted_networks.find do |n|
+          n.name == old_ovs.name.sub(/\A#{Regexp.escape("br-")}/, "")
+        end
+        raise "The current conduit mapping for network '#{network.name}' requires " \
+          "interface '#{our_iface.name}' to be attached to ovs-brige '#{bridge}'. " \
+          "But network '#{other_net.name}' which was already set up, requires " \
+          "the same interface to be attached to ovs-bridge '#{old_ovs.name}'. " \
+          "Refusing to reconfigure. There seems to be a conflict in the conduit" \
+          "mappings for the networks '#{other_net.name}' and '#{network.name}'."
+      end
+
+      # We didn't create the attachement during the current run of chef-client.
+      # Tear down the attachement (and bridge configuration) to be able to attach
+      # the current interface to the new bridge.
+      Chef::Log.warn("Current conduit mapping for network '#{network.name}' requires " \
+                     "ovs-brige '#{bridge}' to be attached to device '#{our_iface.name}' " \
+                     "but the device is already attached to '#{old_ovs.name}'.")
+      Chef::Log.warn("Unplugging '#{our_iface.name}' from '#{old_ovs.name}' and " \
+                     "taking the bridge down for reconfiguration to reduce the " \
+                     "risk of creating a network loop.")
+      old_ovs.unplug(our_iface.name)
+      ::Kernel.system("wicked ifdown #{old_ovs.name}")
+      kill_nic_files old_ovs
+    end
+
     # This flag is used later to enable wicked-nanny (on SUSE platforms)
     ovs_bridge_created = true
 
+    if Nic.exists?(bridge) && Nic.ovs_bridge?(bridge)
+      ovs = Nic.new bridge
+      ovs_slaves = ovs.slaves
+
+      # Find out if the ovs bridge for the current network is already attached to
+      # any other physical, vlan or bond interface than what the current configuration
+      # requires. If it is, tear down the bridge and slave devices for reconfig to
+      # reduce possible risk creating network loops.
+      non_virtual_slaves = ovs_slaves.find_all do |slave|
+        next if slave.name == our_iface.name
+        Nic.base_interface? slave
+      end
+      unless non_virtual_slaves.nil? || non_virtual_slaves.empty?
+        Chef::Log.warn("Current conduit mapping for network '#{network.name}' requires " \
```

Comment:
Same as above. The user may miss that warning and thus wonder why that bridge ceased working. With a failed barclamp that persists this warning (and requires some sort of manual action to retroactively sign off on this bridge removal) that couldn't happen.

Reply:
All this code it just there to keep the bridge working (it will be created further down) after some change in either the network proposal, the node HW  or role assignment of the node was done.